addons/common/functions/fnc_runQuickTimeEvent.sqf

```diff
@@ -0,0 +1,87 @@
+#include "..\script_component.hpp"
+/*
+	 * Author: john681611
+	 * Runs a Quick time Event.
+	 *
+	 * Argument: CodeLength (pos Number)
+	 * 0: Object - where are you inputting values to <OBJECT>
+	 * 1: String - command to execute <STRING>
+	 * 2: Array - command parameters <ARRAY>
+	 * 3: String - command execution code <STRING>
+	 * 4: Array - keyCode <ARRAY>
+	 *
+	 * Return Value:
+	 * None
+	 *
+	 * Example:
+	 * [car, "hint", "Yay", 1, ["↑", "↓", "→", "←"]] call ace_common_fnc_runQuickTimeEvent
+	 *
+	 * Public: Yes
+ */
+
+params ["_veh", "_command", "_command_params", "_command_exec_code", "_keyCode"];
+
+keyCode_keyCode = _keyCode;
+keyCode_done = false;
+keyCode_keyHistory = [];
+hint format ["Command (Hold CTRL down): \n %1", [keyCode_keyCode] call FUNC(getFormattedQuickTimeEventCode)];
+
+keyCode_eventHandler = findDisplay 46 displayAddEventHandler ["KeyUp", {
+	params ["_displayOrControl", "_key", "_shift", "_ctrl", "_alt"];
+	private _override = false;
+	if (!(_key in [DIK_UP, DIK_DOWN, DIK_LEFT, DIK_RIGHT])) exitWith {
+		_override;
+	};
+
+	if (keyCode_done) exitWith {
+		_override;
+	};
+
+	if (!_ctrl) exitWith {
+		keyCode_keyHistory = [];
+		hintSilent format ["Command (Hold CTRL down): \n %1 \n %2", [keyCode_keyCode] call FUNC(getFormattedQuickTimeEventCode), [keyCode_keyHistory] call FUNC(getFormattedQuickTimeEventCode)];
+		_override;
+	};
+	_override = true;
+
+	if (_key isEqualTo DIK_UP) then {
+		keyCode_keyHistory pushBack "↑";
+	};
+
+	if (_key isEqualTo DIK_DOWN) then {
+		keyCode_keyHistory pushBack "↓";
+	};
+
+	if (_key isEqualTo DIK_LEFT) then {
+		keyCode_keyHistory pushBack "←";
+	};
+
+	if (_key isEqualTo DIK_RIGHT) then {
+		keyCode_keyHistory pushBack "→";
+	};
+	if (keyCode_keyHistory isEqualTo keyCode_keyCode) exitWith {
+		hint "Accepted!";
+		keyCode_done = true;
+		keyCode_keyHistory = [];
+		_override;
+	};
+
+	if (!(keyCode_keyHistory isEqualTo (keyCode_keyCode select [0, count keyCode_keyHistory]))) then {
+		keyCode_keyHistory = [];
+	};
+
+	hintSilent format ["Command (Hold CTRL down): \n %1 \n %2", [keyCode_keyCode] call FUNC(getFormattedQuickTimeEventCode), [keyCode_keyHistory] call FUNC(getFormattedQuickTimeEventCode)];
+	_override;
+}];
+
+waitUntil {
```

Comment:
Scheduled code is forbidden in ACE.

Reply:
I'm aware. This is the challenge of moving from a mission script to the mod. If you have suggestions I'm all ears.